### ISSUE TITLE
Unpin werkzeug

### ISSUE THREAD

**Issue by mgasner** [dagster-io/dagster#2119]
Airflow has a dependency on flask-wtf, which has `from werkzeug import url_encode` -- this was marked deprecated in werkzeug 0.16 and errors in werkzeug 1.0.0. See: https://github.com/lepture/flask-wtf/pull/378, https://github.com/lepture/flask-wtf/issues/394

**Comment by karnaker** [comment]
I am seeing a similar error (different application): 
flask-wtf 0.14.2
wekzeug 1.0.0
Hope below helps

Traceback (most recent call last):
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\thewebsite.py", line 4, in <module>
    app = create_app()
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\app\__init__.py", line 40, in create_app
    from app.auth import bp as auth_bp
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\app\auth\__init__.py", line 5, in <module>
    from app.auth import routes
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\app\auth\routes.py", line 8, in <module>
    from app.auth.forms import (LoginForm, SignUpForm, ResetPasswordRequestForm,
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\app\auth\forms.py", line 1, in <module>
    from flask_wtf import FlaskForm
  File "C:\Users\vkarn\.conda\envs\mvp_web_5\Lib\site-packages\flask_wtf\__init__.py", line 17, in <module>
    from .recaptcha import *
  File "C:\Users\vkarn\.conda\envs\mvp_web_5\Lib\site-packages\flask_wtf\recaptcha\__init__.py", line 2, in <module>
    from .fields import *
  File "C:\Users\vkarn\.conda\envs\mvp_web_5\Lib\site-packages\flask_wtf\recaptcha\fields.py", line 3, in <module>
    from . import widgets
  File "C:\Users\vkarn\.conda\envs\mvp_web_5\Lib\site-packages\flask_wtf\recaptcha\widgets.py", line 5, in <module>
    from werkzeug import url_encode
ImportError: cannot import name 'url_encode' from 'werkzeug' (c:\users\vkarn\.conda\envs\mvp_web_5\lib\site-packages\werkzeug\__init__.py)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\vkarn\.conda\envs\mvp_web_5\Lib\site-packages\flask\cli.py", line 343, in __call__
    rv = self._load_unlocked()
  File "C:\Users\vkarn\.conda\envs\mvp_web_5\Lib\site-packages\flask\cli.py", line 330, in _load_unlocked
    self._app = rv = self.loader()
  File "C:\Users\vkarn\.conda\envs\mvp_web_5\Lib\site-packages\flask\cli.py", line 388, in load_app
    app = locate_app(self, import_name, name)
  File "C:\Users\vkarn\.conda\envs\mvp_web_5\Lib\site-packages\flask\cli.py", line 245, in locate_app
    raise NoAppException(
flask.cli.NoAppException: While importing "thewebsite", an ImportError was raised:

Traceback (most recent call last):
  File "c:\users\vkarn\.conda\envs\mvp_web_5\lib\site-packages\flask\cli.py", line 240, in locate_app
    __import__(module_name)
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\thewebsite.py", line 4, in <module>
    app = create_app()
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\app\__init__.py", line 40, in create_app
    from app.auth import bp as auth_bp
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\app\auth\__init__.py", line 5, in <module>
    from app.auth import routes
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\app\auth\routes.py", line 8, in <module>
    from app.auth.forms import (LoginForm, SignUpForm, ResetPasswordRequestForm,
  File "C:\Users\vkarn\OneDrive\repos6\mvp1wk_web\app\auth\forms.py", line 1, in <module>
    from flask_wtf import FlaskForm
  File "c:\users\vkarn\.conda\envs\mvp_web_5\lib\site-packages\flask_wtf\__init__.py", line 17, in <module>
    from .recaptcha import *
  File "c:\users\vkarn\.conda\envs\mvp_web_5\lib\site-packages\flask_wtf\recaptcha\__init__.py", line 2, in <module>
    from .fields import *
  File "c:\users\vkarn\.conda\envs\mvp_web_5\lib\site-packages\flask_wtf\recaptcha\fields.py", line 3, in <module>
    from . import widgets
  File "c:\users\vkarn\.conda\envs\mvp_web_5\lib\site-packages\flask_wtf\recaptcha\widgets.py", line 5, in <module>
    from werkzeug import url_encode
ImportError: cannot import name 'url_encode' from 'werkzeug' (c:\users\vkarn\.conda\envs\mvp_web_5\lib\site-packages\werkzeug\__init__.py)

**Comment by davidism** [comment]
Flask-WTF 0.14.3 is available on PyPI now.

**Comment by schrockn** [comment]
Thanks for following up on this issue! 🙏🏻

**Comment by schrockn** [comment]
Next issue down the dependency graph: https://github.com/flask-admin/flask-admin/issues/1951

Fixed here but unreleased: https://github.com/flask-admin/flask-admin/pull/1922

**Comment by ddelange** [comment]
Just for fun @mgasner @schrockn following the dependency tree: without pin in flask itself, https://github.com/pallets/flask/pull/3489, you will need edge casing of many more flask-dependent packages:

```
$ cat ./pipgrip.lock | grep flask
flask-admin==1.5.4
flask==1.1.1
flask-appbuilder==2.2.2
flask-babel==0.12.2
flask-jwt-extended==3.24.1
flask-login==0.4.1
flask-openid==1.2.5
flask-sqlalchemy==2.4.1
flask-wtf==0.14.3
flask-bcrypt==0.7.1
flask-caching==1.3.3
flask-oauthlib==0.9.5
flask-swagger==0.2.13
```


```
$ pipgrip --tree --lock -vvv apache-airflow[password,crypto,google_auth,postgres,celery,kubernetes,slack,s3]==1.10.8

apache-airflow[celery,crypto,google_auth,kubernetes,password,postgres,s3,slack]==1.10.8 (1.10.8)
├── alembic<2.0,>=1.0 (1.4.0)
│   ├── mako (1.1.1)
│   │   └── markupsafe>=0.9.2 (1.1.1)
│   ├── python-dateutil (2.8.1)
│   │   └── six>=1.5 (1.14.0)
│   ├── python-editor>=0.3 (1.0.4)
│   └── sqlalchemy>=1.1.0 (1.3.13)
├── argcomplete~=1.10 (1.11.1)
│   └── importlib-metadata<2,>=0.23 (1.5.0)
│       └── zipp>=0.5 (2.1.0)
├── attrs~=19.3 (19.3.0)
├── bcrypt>=2.0.0 (3.1.7)
│   ├── cffi>=1.1 (1.13.2)
│   │   └── pycparser (2.19)
│   └── six>=1.4.1 (1.14.0)
├── boto3~=1.10 (1.11.12)
│   ├── botocore<1.15.0,>=1.14.12 (1.14.12)
│   │   ├── docutils<0.16,>=0.10 (0.15.2)
│   │   ├── jmespath<1.0.0,>=0.7.1 (0.9.4)
│   │   ├── python-dateutil<3.0.0,>=2.1 (2.8.1)
│   │   │   └── six>=1.5 (1.14.0)
│   │   └── urllib3<1.26,>=1.20 (1.25.8)
│   ├── jmespath<1.0.0,>=0.7.1 (0.9.4)
│   └── s3transfer<0.4.0,>=0.3.0 (0.3.3)
│       └── botocore<2.0a.0,>=1.12.36 (1.14.12)
│           ├── docutils<0.16,>=0.10 (0.15.2)
│           ├── jmespath<1.0.0,>=0.7.1 (0.9.4)
│           ├── python-dateutil<3.0.0,>=2.1 (2.8.1)
│           │   └── six>=1.5 (1.14.0)
│           └── urllib3<1.26,>=1.20 (1.25.8)
├── cached-property~=1.5 (1.5.1)
├── cattrs~=0.9 (0.9.0)
│   └── attrs>=17.3 (19.3.0)
├── celery~=4.3 (4.4.0)
│   ├── billiard<4.0,>=3.6.1 (3.6.2.0)
│   ├── kombu<4.7,>=4.6.7 (4.6.7)
│   │   ├── amqp<2.6,>=2.5.2 (2.5.2)
│   │   │   └── vine<5.0.0a1,>=1.1.3 (1.3.0)
│   │   └── importlib-metadata>=0.18 (1.5.0)
│   │       └── zipp>=0.5 (2.1.0)
│   ├── pytz>dev (2019.3)
│   └── vine==1.3.0 (1.3.0)
├── colorlog==4.0.2 (4.0.2)
├── configparser<3.6.0,>=3.5.0 (3.5.3)
├── croniter<0.4,>=0.3.17 (0.3.31)
│   └── python-dateutil (2.8.1)
│       └── six>=1.5 (1.14.0)
├── cryptography>=0.9.3 (2.8)
│   ├── cffi!=1.11.3,>=1.8 (1.13.2)
│   │   └── pycparser (2.19)
│   └── six>=1.4.1 (1.14.0)
├── cryptography>=2.0.0 (2.8)
│   ├── cffi!=1.11.3,>=1.8 (1.13.2)
│   │   └── pycparser (2.19)
│   └── six>=1.4.1 (1.14.0)
├── dill<0.4,>=0.2.2 (0.3.1.1)
├── flask-admin==1.5.4 (1.5.4)
│   ├── flask>=0.7 (1.1.1)
│   │   ├── click>=5.1 (7.0)
│   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   └── werkzeug>=0.15 (1.0.0)
│   └── wtforms (2.2.1)
├── flask-appbuilder~=2.2 (2.2.2)
│   ├── apispec[yaml]<2,>=1.1.1 (1.3.3)
│   │   └── pyyaml>=3.10 (5.3)
│   ├── click<8,>=6.7 (7.0)
│   ├── colorama<1,>=0.3.9 (0.4.3)
│   ├── flask-babel<1,>=0.11.1 (0.12.2)
│   │   ├── babel>=2.3 (2.8.0)
│   │   │   └── pytz>=2015.7 (2019.3)
│   │   ├── flask (1.1.1)
│   │   │   ├── click>=5.1 (7.0)
│   │   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   │   └── werkzeug>=0.15 (1.0.0)
│   │   └── jinja2>=2.5 (2.10.3)
│   │       └── markupsafe>=0.23 (1.1.1)
│   ├── flask-jwt-extended<4,>=3.18 (3.24.1)
│   │   ├── flask>=1.0 (1.1.1)
│   │   │   ├── click>=5.1 (7.0)
│   │   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   │   └── werkzeug>=0.15 (1.0.0)
│   │   ├── pyjwt>=1.6.4 (1.7.1)
│   │   ├── six (1.14.0)
│   │   └── werkzeug>=0.14 (1.0.0)
│   ├── flask-login<0.5,>=0.3 (0.4.1)
│   │   └── flask (1.1.1)
│   │       ├── click>=5.1 (7.0)
│   │       ├── itsdangerous>=0.24 (1.1.0)
│   │       ├── jinja2>=2.10.1 (2.10.3)
│   │       │   └── markupsafe>=0.23 (1.1.1)
│   │       └── werkzeug>=0.15 (1.0.0)
│   ├── flask-openid<2,>=1.2.5 (1.2.5)
│   │   ├── flask>=0.10.1 (1.1.1)
│   │   │   ├── click>=5.1 (7.0)
│   │   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   │   └── werkzeug>=0.15 (1.0.0)
│   │   └── python3-openid>=2.0 (3.1.0)
│   │       └── defusedxml (0.6.0)
│   ├── flask-sqlalchemy<3,>=2.4 (2.4.1)
│   │   ├── flask>=0.10 (1.1.1)
│   │   │   ├── click>=5.1 (7.0)
│   │   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   │   └── werkzeug>=0.15 (1.0.0)
│   │   └── sqlalchemy>=0.8.0 (1.3.13)
│   ├── flask-wtf<1,>=0.14.2 (0.14.3)
│   │   ├── flask (1.1.1)
│   │   │   ├── click>=5.1 (7.0)
│   │   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   │   └── werkzeug>=0.15 (1.0.0)
│   │   ├── itsdangerous (1.1.0)
│   │   └── wtforms (2.2.1)
│   ├── flask<2,>=0.12 (1.1.1)
│   │   ├── click>=5.1 (7.0)
│   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   └── werkzeug>=0.15 (1.0.0)
│   ├── jsonschema<4,>=3.0.1 (3.2.0)
│   │   ├── attrs>=17.4.0 (19.3.0)
│   │   ├── importlib-metadata (1.5.0)
│   │   │   └── zipp>=0.5 (2.1.0)
│   │   ├── pyrsistent>=0.14.0 (0.15.7)
│   │   │   └── six (1.14.0)
│   │   ├── setuptools (45.1.0)
│   │   └── six>=1.11.0 (1.14.0)
│   ├── marshmallow-enum<2,>=1.4.1 (1.5.1)
│   │   └── marshmallow>=2.0.0 (2.19.5)
│   ├── marshmallow-sqlalchemy<1,>=0.16.1 (0.21.0)
│   │   ├── marshmallow>=2.15.2 (2.19.5)
│   │   └── sqlalchemy>=1.2.0 (1.3.13)
│   ├── marshmallow<2.20,>=2.18.0 (2.19.5)
│   ├── prison<1.0.0,>=0.1.2 (0.1.2)
│   │   └── six (1.14.0)
│   ├── pyjwt>=1.7.1 (1.7.1)
│   ├── python-dateutil<3,>=2.3 (2.8.1)
│   │   └── six>=1.5 (1.14.0)
│   └── sqlalchemy-utils<1,>=0.32.21 (0.36.1)
│       ├── six (1.14.0)
│       └── sqlalchemy>=1.0 (1.3.13)
├── flask-bcrypt>=0.7.1 (0.7.1)
│   ├── bcrypt (3.1.7)
│   │   ├── cffi>=1.1 (1.13.2)
│   │   │   └── pycparser (2.19)
│   │   └── six>=1.4.1 (1.14.0)
│   └── flask (1.1.1)
│       ├── click>=5.1 (7.0)
│       ├── itsdangerous>=0.24 (1.1.0)
│       ├── jinja2>=2.10.1 (2.10.3)
│       │   └── markupsafe>=0.23 (1.1.1)
│       └── werkzeug>=0.15 (1.0.0)
├── flask-caching<1.4.0,>=1.3.3 (1.3.3)
│   ├── flask (1.1.1)
│   │   ├── click>=5.1 (7.0)
│   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   └── werkzeug>=0.15 (1.0.0)
│   └── werkzeug>=0.12 (1.0.0)
├── flask-login<0.5,>=0.3 (0.4.1)
│   └── flask (1.1.1)
│       ├── click>=5.1 (7.0)
│       ├── itsdangerous>=0.24 (1.1.0)
│       ├── jinja2>=2.10.1 (2.10.3)
│       │   └── markupsafe>=0.23 (1.1.1)
│       └── werkzeug>=0.15 (1.0.0)
├── flask-oauthlib>=0.9.1 (0.9.5)
│   ├── flask (1.1.1)
│   │   ├── click>=5.1 (7.0)
│   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   └── werkzeug>=0.15 (1.0.0)
│   ├── oauthlib!=2.0.3,!=2.0.4,!=2.0.5,<3.0.0,>=1.1.2 (2.1.0)
│   └── requests-oauthlib>=0.6.2 (1.1.0)
│       ├── oauthlib<3.0.0,>=2.1.0 (2.1.0)
│       └── requests>=2.0.0 (2.22.0)
│           ├── certifi>=2017.4.17 (2019.11.28)
│           ├── chardet<3.1.0,>=3.0.2 (3.0.4)
│           ├── idna<2.9,>=2.5 (2.8)
│           └── urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (1.25.8)
├── flask-swagger==0.2.13 (0.2.13)
│   ├── flask>=0.10 (1.1.1)
│   │   ├── click>=5.1 (7.0)
│   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   └── werkzeug>=0.15 (1.0.0)
│   └── pyyaml>=3.0 (5.3)
├── flask-wtf<0.15,>=0.14.2 (0.14.3)
│   ├── flask (1.1.1)
│   │   ├── click>=5.1 (7.0)
│   │   ├── itsdangerous>=0.24 (1.1.0)
│   │   ├── jinja2>=2.10.1 (2.10.3)
│   │   │   └── markupsafe>=0.23 (1.1.1)
│   │   └── werkzeug>=0.15 (1.0.0)
│   ├── itsdangerous (1.1.0)
│   └── wtforms (2.2.1)
├── flask<2.0,>=1.1.0 (1.1.1)
│   ├── click>=5.1 (7.0)
│   ├── itsdangerous>=0.24 (1.1.0)
│   ├── jinja2>=2.10.1 (2.10.3)
│   │   └── markupsafe>=0.23 (1.1.1)
│   └── werkzeug>=0.15 (1.0.0)
├── flower<1.0,>=0.7.3 (0.9.3)
│   ├── babel<3.0.0,>=1.0 (2.8.0)
│   │   └── pytz>=2015.7 (2019.3)
│   ├── celery>=3.1.0 (4.4.0)
│   │   ├── billiard<4.0,>=3.6.1 (3.6.2.0)
│   │   ├── kombu<4.7,>=4.6.7 (4.6.7)
│   │   │   ├── amqp<2.6,>=2.5.2 (2.5.2)
│   │   │   │   └── vine<5.0.0a1,>=1.1.3 (1.3.0)
│   │   │   └── importlib-metadata>=0.18 (1.5.0)
│   │   │       └── zipp>=0.5 (2.1.0)
│   │   ├── pytz>dev (2019.3)
│   │   └── vine==1.3.0 (1.3.0)
│   ├── pytz (2019.3)
│   └── tornado<6.0.0,>=4.2.0 (5.1.1)
├── funcsigs<2.0.0,>=1.0.0 (1.0.2)
├── future<0.17,>=0.16.0 (0.16.0)
├── graphviz>=0.12 (0.13.2)
├── gunicorn<20.0,>=19.5.0 (19.10.0)
├── iso8601>=0.1.12 (0.1.12)
├── jinja2<2.11.0,>=2.10.1 (2.10.3)
│   └── markupsafe>=0.23 (1.1.1)
├── json-merge-patch==0.2 (0.2)
├── jsonschema~=3.0 (3.2.0)
│   ├── attrs>=17.4.0 (19.3.0)
│   ├── importlib-metadata (1.5.0)
│   │   └── zipp>=0.5 (2.1.0)
│   ├── pyrsistent>=0.14.0 (0.15.7)
│   │   └── six (1.14.0)
│   ├── setuptools (45.1.0)
│   └── six>=1.11.0 (1.14.0)
├── kubernetes>=3.0.0 (10.0.1)
│   ├── certifi>=14.05.14 (2019.11.28)
│   ├── google-auth>=1.0.1 (1.11.0)
│   │   ├── cachetools<5.0,>=2.0.0 (4.0.0)
│   │   ├── pyasn1-modules>=0.2.1 (0.2.8)
│   │   │   └── pyasn1<0.5.0,>=0.4.6 (0.4.8)
│   │   ├── rsa<4.1,>=3.1.4 (4.0)
│   │   │   └── pyasn1>=0.1.3 (0.4.8)
│   │   ├── setuptools>=40.3.0 (45.1.0)
│   │   └── six>=1.9.0 (1.14.0)
│   ├── python-dateutil>=2.5.3 (2.8.1)
│   │   └── six>=1.5 (1.14.0)
│   ├── pyyaml>=3.12 (5.3)
│   ├── requests (2.22.0)
│   │   ├── certifi>=2017.4.17 (2019.11.28)
│   │   ├── chardet<3.1.0,>=3.0.2 (3.0.4)
│   │   ├── idna<2.9,>=2.5 (2.8)
│   │   └── urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (1.25.8)
│   ├── requests-oauthlib (1.1.0)
│   │   ├── oauthlib<3.0.0,>=2.1.0 (2.1.0)
│   │   └── requests>=2.0.0 (2.22.0)
│   │       ├── certifi>=2017.4.17 (2019.11.28)
│   │       ├── chardet<3.1.0,>=3.0.2 (3.0.4)
│   │       ├── idna<2.9,>=2.5 (2.8)
│   │       └── urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (1.25.8)
│   ├── setuptools>=21.0.0 (45.1.0)
│   ├── six>=1.9.0 (1.14.0)
│   ├── urllib3>=1.24.2 (1.25.8)
│   └── websocket-client!=0.40.0,!=0.41.*,!=0.42.*,>=0.32.0 (0.54.0)
│       └── six (1.14.0)
├── lazy-object-proxy~=1.3 (1.4.3)
├── markdown<3.0,>=2.5.2 (2.6.11)
├── oauthlib!=2.0.3,!=2.0.4,!=2.0.5,<3.0.0,>=1.1.2 (2.1.0)
├── pandas<1.0.0,>=0.17.1 (0.25.3)
│   ├── numpy>=1.13.3 (1.18.1)
│   ├── python-dateutil>=2.6.1 (2.8.1)
│   │   └── six>=1.5 (1.14.0)
│   └── pytz>=2017.2 (2019.3)
├── pendulum==1.4.4 (1.4.4)
│   ├── python-dateutil<3.0.0.0,>=2.6.0.0 (2.8.1)
│   │   └── six>=1.5 (1.14.0)
│   ├── pytzdata>=2018.3.0.0 (2019.3)
│   └── tzlocal<2.0.0.0,>=1.5.0.0 (1.5.1)
│       └── pytz (2019.3)
├── psutil<6.0.0,>=4.2.0 (5.6.7)
├── psycopg2-binary>=2.7.4 (2.8.4)
├── pygments<3.0,>=2.0.1 (2.5.2)
├── python-daemon<2.2,>=2.1.1 (2.1.2)
│   ├── docutils (0.15.2)
│   ├── lockfile>=0.10 (0.12.2)
│   └── setuptools (45.1.0)
├── python-dateutil<3,>=2.3 (2.8.1)
│   └── six>=1.5 (1.14.0)
├── requests-oauthlib==1.1.0 (1.1.0)
│   ├── oauthlib<3.0.0,>=2.1.0 (2.1.0)
│   └── requests>=2.0.0 (2.22.0)
│       ├── certifi>=2017.4.17 (2019.11.28)
│       ├── chardet<3.1.0,>=3.0.2 (3.0.4)
│       ├── idna<2.9,>=2.5 (2.8)
│       └── urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (1.25.8)
├── requests<3,>=2.20.0 (2.22.0)
│   ├── certifi>=2017.4.17 (2019.11.28)
│   ├── chardet<3.1.0,>=3.0.2 (3.0.4)
│   ├── idna<2.9,>=2.5 (2.8)
│   └── urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (1.25.8)
├── setproctitle<2,>=1.1.8 (1.1.10)
├── slackclient<2.0.0,>=1.0.0 (1.3.2)
│   ├── requests<3.0a0,>=2.11 (2.22.0)
│   │   ├── certifi>=2017.4.17 (2019.11.28)
│   │   ├── chardet<3.1.0,>=3.0.2 (3.0.4)
│   │   ├── idna<2.9,>=2.5 (2.8)
│   │   └── urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 (1.25.8)
│   ├── six<2.0a0,>=1.10 (1.14.0)
│   └── websocket-client<0.55.0,>=0.35 (0.54.0)
│       └── six (1.14.0)
├── sqlalchemy-jsonfield~=0.9 (0.9.0)
│   └── sqlalchemy (1.3.13)
├── sqlalchemy~=1.3 (1.3.13)
├── tabulate<0.9,>=0.7.5 (0.8.6)
├── tenacity==4.12.0 (4.12.0)
│   └── six>=1.9.0 (1.14.0)
├── termcolor==1.1.0 (1.1.0)
├── text-unidecode==1.2 (1.2)
├── thrift>=0.9.2 (0.13.0)
│   └── six>=1.7.2 (1.14.0)
├── tornado<6.0,>=4.2.0 (5.1.1)
├── typing-extensions>=3.7.4 (3.7.4.1)
├── tzlocal<2.0.0,>=1.4 (1.5.1)
│   └── pytz (2019.3)
├── unicodecsv>=0.14.1 (0.14.1)
```

**Comment by davidism** [comment]
We're planning to add compatibility imports back, if you have insight into what specific imports raise warnings with werkzeug 0.16.1 that would be an amazing help. Thanks for your patience.

**Comment by ddelange** [comment]
`secure_filename` would be nice, it was breaking our airflow 1.10.8 installation (fixed for 1.10.9/2 https://github.com/apache/airflow/pull/7377). for the rest i think it'll be a tough one to find out which ones actually occur oftren. i'm curious if there's some github crawling/statistics software for this purpose, since github search doesn't cut it... https://github.com/search?o=desc&p=2&q=%22from+werkzeug+import%22&s=indexed&type=Code

```
File "/usr/local/lib/python3.7/site-packages/flask_admin/model/__init__.py", line 2, in <module>
from .base import BaseModelView
File "/usr/local/lib/python3.7/site-packages/flask_admin/model/base.py", line 8, in <module>
from werkzeug import secure_filename
ImportError: cannot import name 'secure_filename' from 'werkzeug' (/usr/local/lib/python3.7/site-packages/werkzeug/__init__.py)
```
